src/ERC20.sol

```diff
@@ -29,15 +29,15 @@ abstract contract ERC20 is State {
         return true;
     }
 
-    function transferFrom(address sender, address recipient, uint256 amount) external isFunctionAllowed nonReentrant returns (bool) {
+    function transferFrom(address sender, address recipient, uint256 amount) external isFunctionAllowed isReceiverWhitelisted(recipient) nonReentrant returns (bool) {
         allowance[sender][msg.sender] -= amount;
         balanceOf[sender] -= amount;
         balanceOf[recipient] += amount;
         emit Transfer(sender, recipient, amount);
         return true;
     }
 
-    function _mint(address to, uint256 amount) internal {
+    function _mint(address to, uint256 amount) isReceiverWhitelisted(to) internal  {
```

Comment:
You only whitelist mint. We heed whitelist anyone who can recive Borrow and Collateral also